packages/gatsby/src/redux/actions.js

```diff
@@ -195,6 +198,39 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
       console.log(``)
       console.log(page)
       noPageOrComponent = true
+    } else if (page.component) {
+      // normalize component path
+      page.component = slash(page.component)
+      // check if path uses correct casing - incorrect casing will
+      // casue issues in query compiler and incosistencies when
+      // developing on Mac or Windows and trying to deploy from
+      // linux CI/CD pipeline
+      const trueComponentPath = slash(truePath(page.component))
+      if (trueComponentPath !== page.component) {
+        if (!hasWarnedForPageComponentInvalidCasing.has(page.component)) {
+          const markers = page.component
+            .split(``)
+            .map((letter, index) => {
+              if (letter !== trueComponentPath[index]) {
+                return `^`
+              }
+              return ` `
+            })
+            .join(``)
+
+          report.warn(
+            stripIndent`
+          ${name} created a page with a component path that doesn't match casing of file. This will work locally, but will break on Linux (on most CI/CD pipelines).
```

Comment:
```suggestion
          ${name} created a page with a component path that doesn't match the casing of the actual file. This may work locally, but will break on systems which are case-sensitive, e.g. most CI/CD pipelines.
```

Reply:
nice syntax highlighting ☝️  haha

Reply:
Do we want to add a gatsby.app (or gatsby.dev) short link for this? Worth documenting?

Reply:
I think this hopefully is more selfexplaining than async `createPages` warning, so I would personally skip creating dedicated page for it (we just try to adjust file casing and pointing where exactly it needs to change)

Reply:
Yeah - agreed. I've been thinking it may be a helpful thing to always have a documentation page for errors, but perhaps that's overkill (especially when the error message is self-documenting, as it is here);.